### PR TITLE
Add $bit update operator tests

### PR DESCRIPTION
**Pull request by imforster**
#387

Add comprehensive tests for the $bit update operator covering:
- Core AND/OR/XOR operations with boundary values and two's complement
- Type preservation and Int32/Int64 cross-type promotion
- Field path resolution, missing field initialization, and multi-field updates
- Result metadata (n/nModified) and upsert behavior
- Positional array operators ($, $[], $[elem])
- Error cases: field type rejection, operand type rejection, operation key validation, empty sub-doc, path conflicts, and type change after $set

**108 test cases** across 7 files following project test guidelines.

## File Structure

| File | Tests | Covers |
|---|---|---|
| test_bit_core_operations.py | 46 | AND/OR/XOR, boundaries, two's complement, negatives, bit patterns |
| test_bit_type_promotion.py | 16 | Int32/Int64 preservation, cross-type promotion, dataset tests |
| test_bit_field_paths.py | 8 | Dot notation, missing fields, array index, multiple fields |
| test_bit_update_methods.py | 7 | Result metadata (n/nModified), upsert behavior |
| test_bit_array_positional.py | 3 | $, $[], $[elem] positional operators |
| test_bit_error_cases.py | 40 | Type rejection, operation keys, path conflicts, type change |
| test_smoke_bit.py | 1 | Basic smoke test (pre-existing) |

### CODE REVIEW

File: documentdb_tests/compatibility/tests/core/operator/update/bitwise/bit/test_bit_error_cases.py [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+@pytest.mark.parametrize("test", pytest_params(OPERATION_KEY_ERROR_TESTS))
+def test_bit_operation_key_errors(collection, test: BitErrorTest):
+    """Test $bit rejects invalid operation keys."""
+    result = run_bit_error(collection, test)
+    assertFailureCode(result, test.error_code, msg=test.msg)  # type: ignore[arg-type]
+
+
+# Property [Path Conflicts]: $bit rejects conflicting update paths and conflicts with
+# other update operators on the same field.
+CONFLICTING_UPDATE_OPERATORS_ERROR = 40
```
Comment (eerxuan) [inline documentdb_tests/compatibility/tests/core/operator/update/bitwise/bit/test_bit_error_cases.py]: This constant already exists at framework/error_codes.py:15. Inline redefinition violates the project's "use constants from framework.error_codes" rule. Replace with import. The rest of
  the file imports BAD_VALUE_ERROR and FAILED_TO_PARSE_ERROR correctly — this one was missed.

File: documentdb_tests/compatibility/tests/core/operator/update/bitwise/bit/test_bit_error_cases.py [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+        collection,
+        {
+            "update": collection.name,
+            "updates": [{"q": {"_id": 1}, "u": {"$bit": {"arr.0": {"and": 1}}}}],
+        },
+    )
+    assertFailureCode(result, BAD_VALUE_ERROR, msg="$bit should reject non-integer array element.")
+
+
+def test_bit_after_type_change_to_string(collection):
```
Comment (eerxuan) [inline documentdb_tests/compatibility/tests/core/operator/update/bitwise/bit/test_bit_error_cases.py]: 🟠 Major — test_bit_conflict_with_unset_same_field, test_bit_after_type_change_to_string and test_bit_succeeds_after_type_change_to_int64 test field-type-at-update-time, not $bit-specific

  test_bit_error_cases.py:380-424. Two tests do:
  1. Insert {v: 10}
  2. $set: {v: <new_type>} to change field type
  3. $bit operates on the new type

  These test that $bit validates field type at update execution time (not at command parse time). But that's a property of any type-strict update operator. If the field starts as Int and
  is $set to a string, any subsequent type-checking update fails — this is generic, not $bit-specific.

  Furthermore, the multi-step setup (two separate $set then $bit updates) tests update-operator sequencing rather than $bit itself. Recommend dropping these or moving to a generic
  "field-type-at-update-time" test in a higher-level update folder.

File: documentdb_tests/compatibility/tests/core/operator/update/bitwise/bit/test_bit_core_operations.py [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+from documentdb_tests.framework.test_case import BaseTestCase
+from documentdb_tests.framework.test_constants import (
+    INT32_MAX,
+    INT32_MIN,
+    INT64_MAX,
+    INT64_MIN,
+)
+
+
+@dataclass(frozen=True)
```
Comment (eerxuan) [inline documentdb_tests/compatibility/tests/core/operator/update/bitwise/bit/test_bit_core_operations.py:20]: 🟡 Minor — 46 cases in test_bit_core_operations.py overlaps with test_bit_type_promotion.py

  The core-operations file has cases like:
  - and_int64_zero_mask, and_int64_low_nibble, or_int64_identity, or_int64_basic, xor_int64_identity, xor_int64_self_produces_zero (6 cases)
  - All boundary boundary_int64_* (4 cases)

  These prove "AND/OR/XOR work with Int64 values." But test_bit_type_promotion.py already covers Int64 + Int64 in int64_stays_int64, int64_or_stays_int64, int64_zero_result_stays_int64
  plus dataset cases.

  Suggest: keep core-operations focused on AND/OR/XOR semantics with Int32 representative cases; let type_promotion cover Int64 type behavior. Trim the *_int64_* core cases.

File: documentdb_tests/compatibility/tests/core/operator/update/bitwise/bit/test_bit_error_cases.py (hunk 1 above)
Comment (imforster) [reply documentdb_tests/compatibility/tests/core/operator/update/bitwise/bit/test_bit_error_cases.py]: Updated to use central error codes.

File: documentdb_tests/compatibility/tests/core/operator/update/bitwise/bit/test_bit_error_cases.py (hunk 2 above)
Comment (imforster) [reply documentdb_tests/compatibility/tests/core/operator/update/bitwise/bit/test_bit_error_cases.py]: Dropped

File: documentdb_tests/compatibility/tests/core/operator/update/bitwise/bit/test_bit_core_operations.py (hunk 3 above)
Comment (imforster) [reply documentdb_tests/compatibility/tests/core/operator/update/bitwise/bit/test_bit_core_operations.py:20]: Trimmed all 10 overlapping Int64 cases from test_bit_core_operations.py:

Removed the 6 Int64 AND/OR/XOR cases (and_int64_zero_mask, and_int64_low_nibble, or_int64_identity, or_int64_basic, xor_int64_identity, xor_int64_self_produces_zero)
Removed the 4 boundary_int64_* cases
Core operations is now purely Int32-focused for AND/OR/XOR semantics. Int64 behavior lives in test_bit_type_promotion.py. That's what took the test count from 118 down to 108.

**Review by eerxuan — APPROVED** [review]
